src/testing/tests/precompiles/ecadd.rs

```diff
@@ -0,0 +1,162 @@
+    let mut memory = SimpleMemory::new();
+    let mut precompiles_processor = DefaultPrecompilesProcessor::<false>;
+    let page_number = 4u32;
+    // create heap page
+    memory.populate_page(vec![
+        (page_number, vec![U256::zero(); 1 << 10]),
+        (page_number + 1, vec![]),
+    ]);
+
+    // fill the memory
+    let num_words_used = fill_memory(x1, y1, x2, y2, page_number, &mut memory);
+
+    let precompile_call_params = PrecompileCallABI {
+        input_memory_offset: 0,
+        input_memory_length: num_words_used as u32,
+        output_memory_offset: num_words_used as u32,
+        output_memory_length: 3,
+        memory_page_to_read: page_number,
+        memory_page_to_write: page_number,
+        precompile_interpreted_data: 0,
+    };
+    let precompile_call_params_encoded = precompile_call_params.to_u256();
+
+    let address = Address::from_low_u64_be(ECADD_INNER_FUNCTION_PRECOMPILE_ADDRESS as u64);
+
+    let precompile_query = LogQuery {
+        timestamp: Timestamp(1u32),
+        tx_number_in_block: 0,
+        shard_id: 0,
+        aux_byte: PRECOMPILE_AUX_BYTE,
+        address,
+        key: precompile_call_params_encoded,
+        read_value: U256::zero(),
+        written_value: U256::zero(),
+        rw_flag: false,
+        rollback: false,
+        is_service: false,
+    };
+
+    let _ = precompiles_processor.execute_precompile(4, precompile_query, &mut memory);
+
+    let range = 0u32..(num_words_used as u32 + 3);
+    let content = memory.dump_page_content(page_number, range.clone());
+    let content_len = content.len();
+    let ok_or_error_marker = content[content_len - 3];
+    let output_x = content[content_len - 2];
+    let output_y = content[content_len - 1];
+
+    if expect_ok {
+        let mut buffer = [0u8; 32];
+        U256::one().to_big_endian(&mut buffer);
+        assert_eq!(ok_or_error_marker, buffer);
+        assert_eq!(&output_x, &expected_x);
+        assert_eq!(&output_y, &expected_y);
+    } else {
+        let mut buffer = [0u8; 32];
+        U256::zero().to_big_endian(&mut buffer);
+        assert_eq!(ok_or_error_marker, buffer);
+        assert_eq!(&output_x[..], &[0u8; 32]);
+        assert_eq!(&output_y[..], &[0u8; 32]);
+    }
+
+    (content, range)
+}
+
+fn ecadd_test_inner_from_raw(
+    raw_input: &str,
+    raw_x: &str,
+    raw_y: &str,
+    expect_ok: bool,
+) -> (Vec<[u8; 32]>, std::ops::Range<u32>) {
+    let input_bytes = hex::decode(raw_input).unwrap();
+    let x1: [u8; 32] = input_bytes[0..32].try_into().unwrap();
+    let y1: [u8; 32] = input_bytes[32..64].try_into().unwrap();
+    let x2: [u8; 32] = input_bytes[64..96].try_into().unwrap();
+    let y2: [u8; 32] = input_bytes[96..128].try_into().unwrap();
+
+    let x: [u8; 32] = hex::decode(raw_x).unwrap().try_into().unwrap();
+    let y: [u8; 32] = hex::decode(raw_y).unwrap().try_into().unwrap();
+
+    ecadd_test_inner(x1, y1, x2, y2, expect_ok, x, y)
+}
+
+#[test]
+fn test_valid() {
```

Comment:
for completeness, should we also try a failing test?

Reply:
Yeah, I added the test where the point on the curve is invalid (meaning it is not on the curve). Seems to work well :)